app/src/androidTest/java/protect/card_locker/LanguageChangeUITest.java

```diff
@@ -0,0 +1,160 @@
+                                0)))
+                .atPosition(8);
+        appCompatCheckedTextView.perform(click());
+
+        ViewInteraction textView = onView(
+                allOf(withText("Einstellungen"),
+                        withParent(allOf(withId(R.id.toolbar),
+                                withParent(IsInstanceOf.<View>instanceOf(android.widget.LinearLayout.class)))),
+                        isDisplayed()));
+        textView.check(matches(withText("Einstellungen")));
+
+        ViewInteraction recyclerView2 = onView(
+                allOf(withId(com.jaredrummler.android.colorpicker.R.id.recycler_view),
+                        childAtPosition(
+                                withId(android.R.id.list_container),
+                                0)));
+        recyclerView2.perform(actionOnItemAtPosition(4, click()));
+
+        DataInteraction appCompatCheckedTextView2 = onData(anything())
+                .inAdapterView(allOf(withId(androidx.appcompat.R.id.select_dialog_listview),
+                        childAtPosition(
+                                withId(androidx.appcompat.R.id.contentPanel),
+                                0)))
+                .atPosition(9);
+        appCompatCheckedTextView2.perform(click());
+
+        ViewInteraction textView2 = onView(
+                allOf(withText("Ρυθμίσεις"),
+                        withParent(allOf(withId(R.id.toolbar),
+                                withParent(IsInstanceOf.<View>instanceOf(android.widget.LinearLayout.class)))),
+                        isDisplayed()));
+        textView2.check(matches(withText("Ρυθμίσεις")));
+
+        ViewInteraction recyclerView3 = onView(
+                allOf(withId(com.jaredrummler.android.colorpicker.R.id.recycler_view),
+                        childAtPosition(
+                                withId(android.R.id.list_container),
+                                0)));
+        recyclerView3.perform(actionOnItemAtPosition(4, click()));
+
+        DataInteraction appCompatCheckedTextView3 = onData(anything())
+                .inAdapterView(allOf(withId(androidx.appcompat.R.id.select_dialog_listview),
+                        childAtPosition(
+                                withId(androidx.appcompat.R.id.contentPanel),
+                                0)))
+                .atPosition(0);
+        appCompatCheckedTextView3.perform(click());
+
+        ViewInteraction appCompatImageButton = onView(
+                allOf(withContentDescription("Navigate up"),
+                        childAtPosition(
+                                allOf(withId(R.id.toolbar),
+                                        childAtPosition(
+                                                withClassName(is("com.google.android.material.appbar.AppBarLayout")),
+                                                0)),
+                                1),
+                        isDisplayed()));
+        appCompatImageButton.perform(click());
```

Comment:
The test fails for me here when testing on an Android 14 emulator:

```
androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: (view.getContentDescription() is "Navigate up" and Child at position 1 in parent (view.getId() is <2131296920> and Child at position 0 in parent view.getClass().getName() matches: is "com.google.android.material.appbar.AppBarLayout") and (view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle))
```

I've attached the full view-hierarchy:
[view-hierarchy-1.txt](https://github.com/user-attachments/files/15521064/view-hierarchy-1.txt)

Reply:
This seems weird to me because I am also running the tests on an Android 14 emulator, but I will examine this further.